spec/support/saml_response_helper.rb

```diff
@@ -18,10 +26,26 @@ def input_id
       'SAMLResponse'
     end
 
-    def response_doc
-      return Nokogiri::XML(xml_response) if @test_type == 'feature'
+    def raw_xml_response
+      if @test_type == 'feature'
+        xml_response
+      else
+        @response.body
+      end
+    end
 
-      Nokogiri::XML(@response.body)
+    def response_doc
+      if raw_xml_response =~ /EncryptedData/
+        @original_encrypted = true
+        Nokogiri::XML(
+          OneLogin::RubySaml::Response.new(
+            raw_xml_response,
+            settings: sp1_saml_settings
+          ).decrypted_document.to_s
+        )
+      else
+        Nokogiri::XML(raw_xml_response)
```

Comment:
nit: Set @original_encrypted = false here rather then in initialize perhaps?  Not sure about ruby style but I would have though that it would better to leave it undefined until you know whether its encrypted or not.